ibm/flex/structures.go

```diff
@@ -3220,3 +3221,51 @@ func FlattenSatelliteClusterZones(list []string) []map[string]interface{} {
 	}
 	return zones
 }
+
+func FetchResourceInstanceDetails(d *schema.ResourceData, meta interface{}, instanceID string) error {
+	// Get ResourceController from ClientSession
+	resourceControllerClient, err := meta.(conns.ClientSession).ResourceControllerV2API()
+	if err != nil {
+		return err
+	}
+
+	getResourceOpts := resourcecontrollerv2.GetResourceInstanceOptions{
+		ID: &instanceID,
+	}
+
+	instance, response, err := resourceControllerClient.GetResourceInstance(&getResourceOpts)
+	if err != nil {
+		log.Printf("[DEBUG] Error retrieving resource instance: %s\n%s", err, response)
+		return fmt.Errorf("Error retrieving resource instance: %s\n%s", err, response)
+	}
+	if strings.Contains(*instance.State, "removed") {
+		log.Printf("[WARN] Removing instance from TF state because it's now in removed state")
+		d.SetId("")
```

Comment:
why is the ID set to empty?
may be we must return the error..

Reply:
FYI, I changed this to return an error so let me know if there is anything else that needs to change in this PR. Otherwise this is good to go